examples/control_flow/parallel_executor/src/nat_parallel_executor/register.py

```
@@ -0,0 +1,130 @@
+                    "%s <- completed branch=%s duration=%.3fs",
+                    log_prefix,
+                    tool_name,
+                    perf_counter() - branch_start,
+                )
+            return result
+
+        for tool_name_ref in config.tool_list:
+            tool_name = str(tool_name_ref)
+            tool = tools_dict.get(tool_name)
+            if tool is None:
+                raise ValueError(f"Parallel executor: unknown tool '{tool_name}'")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)